test/connectors/SDDSConnectorSpec.scala

```diff
@@ -47,12 +47,7 @@ class SDDSConnectorSpec extends SpecBase {
   }
 
   trait Setup {
-    val mockHttpClient: HttpClientV2   = mock[HttpClientV2]
-    val requestBuilder: RequestBuilder = mock[RequestBuilder]
-
-    implicit val hc: HeaderCarrier = HeaderCarrier()
-
-    val application: Application = applicationBuilder()
+    val application: Application = applicationBuilder
```

Comment:
IMO everywhere we have a locally defined `val application: Application` it should override the application in SpecBase. That will make it more readable and another benefit would be that `def instanceOf` defined in SPecBase will always use the correct application instance.

Reply:
Only issue that we define them in Setup :thinking:

Reply:
We could just have Setup extend SpecBase :shrug: